web/extensions/core/maskeditor.js

```diff
@@ -0,0 +1,568 @@
+		if(backupData.data[i+3] == 255)
+			backupData.data[i+3] = 0;
+		else
+			backupData.data[i+3] = 255;
+
+		backupData.data[i] = 0;
+		backupData.data[i+1] = 0;
+		backupData.data[i+2] = 0;
+	}
+
+	backupCtx.globalCompositeOperation = 'source-over';
+	backupCtx.putImageData(backupData, 0, 0);
+}
+
+class MaskEditorDialog extends ComfyDialog {
+	static instance = null;
+	constructor() {
+		super();
+		this.element = $el("div.comfy-modal", { parent: document.body }, 
+			[ $el("div.comfy-modal-content", 
+				[...this.createButtons()]),
+			]);
+		MaskEditorDialog.instance = this;
+	}
+
+	createButtons() {
+		return [];
+	}
+
+	clearMask(self) {
+	}
+
+	createButton(name,callback) {
+		var button = document.createElement("button");
+		button.innerText = name;
+		button.style.position = "absolute";
+		button.style.top = "5px";
+		button.addEventListener("click", callback);
+		return button;
+	}
+
+	createLeftButton(name,left,callback) {
+		var button = this.createButton(name,callback);
+		button.style.left = left;
+		return button;
+	}
+
+	createRightButton(name,right,callback) {
+		var button = this.createButton(name,callback);
+		button.style.right = right;
+		return button;
+	}
+
+	createLeftSlider(self, name,left,callback) {
+		const divElement = document.createElement('div');
+		divElement.style.position = "absolute";
+		divElement.style.top = "5px";
+		divElement.style.left = left;
+		divElement.style.backgroundColor = 'Black';
+		divElement.style.paddingLeft = "10px";
+		divElement.style.paddingRight = "10px";
+		divElement.style.borderRadius = "10px";
+
+		self.brush_slider_input = document.createElement('input');
+		self.brush_slider_input.setAttribute('type', 'range');
+		self.brush_slider_input.setAttribute('min', '1');
+		self.brush_slider_input.setAttribute('max', '100');
+		self.brush_slider_input.setAttribute('value', '10');
+
+		const labelElement = document.createElement("label");
+		labelElement.innerHTML = `<font color='white' size='5px'>${name}</font>`;
+
+		divElement.appendChild(labelElement);
+		divElement.appendChild(self.brush_slider_input);
+
+		self.brush_slider_input.addEventListener("change", callback);
+
+		return divElement;
+	}
+
+	setlayout(imgCanvas, maskCanvas) {
+		const self = this;
+
+		// If it is specified as relative, using it only as a hidden placeholder for padding is recommended
+		// to prevent anomalies where it exceeds a certain size and goes outside of the window.
+		var placeholder = document.createElement("div");
+		placeholder.style.position = "relative";
+		placeholder.style.height = "50px";
+
+		var bottom_panel = document.createElement("div");
+		bottom_panel.style.position = "absolute";
+		bottom_panel.style.bottom = "0px";
+		bottom_panel.style.left = "0";
+		bottom_panel.style.right = "0";
+		bottom_panel.style.height = "50px";
+
+		var brush = document.createElement("div");
+		brush.id = "brush";
+		brush.style.backgroundColor = "transparent";
+		brush.style.backdropFilter = "invert(1)"
```

Comment:
```suggestion
        brush.style.outline = "1px dashed black";
        brush.style.boxShadow = "0 0 0 1px white";
```
Better solution to brush visibility. Alternating black and white, like gimp.